### DISCUSSION TITLE
Color the prompt of `bin/rails console` even on non-standard environments.

### DISCUSSION THREAD

**Discussion by Uaitt** [rails/rails#52677] [Ideas]
When the Rails console is launched in a non-standard environment (e.g. `staging`), the prompt isn't colored. 

![Screenshot 2024-08-22 at 16 24 30](https://github.com/user-attachments/assets/c90e13ae-7f67-4dfb-a7d4-7f126dba99a9)

That's the result of this piece of code: https://github.com/rails/rails/blob/main/railties/lib/rails/commands/console/irb_console.rb#L122-L133

Don't you think it would be better to assign a color to non-standard envs? I know it's subjective, but leaving it not colored just feels very ugly in my opinion 😅.

**Comment by MatheusRich** [comment]
I'd say we should colorize it as well. I wonder if it should be the same color as prod or something else?

**Reply by Uaitt** [reply]
@MatheusRich Actually I wouldn't give it the same color of prod.

This way when I open a console and I see red in the prompt my eyes automatically understand that I am on production and that I should be extra careful before launching commands.

Those are the colors that IRB allows => https://github.com/ruby/irb/blob/master/lib/irb/color.rb#L8-L19

Excluding black, white, blue and red we can choose from:
- green
- yellow
- magenta
- cyan

**Reply by MatheusRich** [reply]
I'd like to experiment with yellow and magenta. They're different enough from dev and prod, but still something that will  catch our attention

**Reply by Uaitt** [reply]
@MatheusRich did you have some time to take a look at this? If so, which color do you like best?

**Reply by MatheusRich** [reply]
I'd say you should pick whatever you prefer and open a PR. Let's discuss details there.

**Reply by Uaitt** [reply]
@MatheusRich I have opened #53027

**Comment by Uaitt** [comment]
https://github.com/rails/rails/pull/53027 has been merged, thus I can close this.
